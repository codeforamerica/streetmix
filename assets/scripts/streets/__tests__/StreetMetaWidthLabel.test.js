/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import StreetMetaWidthLabel from '../StreetMetaWidthLabel'

const formatMessage = jest.fn(({ defaultMessage }) => defaultMessage)
const dummyStreetObject = {
  units: 0,
  width: 10,
  occupiedWidth: 10,
  remainingWidth: 0
}

describe('StreetMetaWidthLabel', () => {
  it('renders when editable', () => {
    const wrapper = shallow(
      <StreetMetaWidthLabel
        formatMessage={formatMessage}
        street={dummyStreetObject}
        editable
        onClick={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders when not editable', () => {
    const wrapper = shallow(
      <StreetMetaWidthLabel
        formatMessage={formatMessage}
        street={dummyStreetObject}
        editable={false}
        onClick={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with remaining width', () => {
    const wrapper = shallow(
      <StreetMetaWidthLabel
        formatMessage={formatMessage}
        street={{
          ...dummyStreetObject,
          occupiedWidth: 9,
          remainingWidth: 1
        }}
        editable
        onClick={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with over available width', () => {
    const wrapper = shallow(
      <StreetMetaWidthLabel
        formatMessage={formatMessage}
        street={{
          ...dummyStreetObject,
          occupiedWidth: 11,
          remainingWidth: -1
        }}
        editable
        onClick={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('calls onClick handler when clicked', () => {
    const handleClick = jest.fn()
    const wrapper = shallow(
      <StreetMetaWidthLabel
        formatMessage={formatMessage}
        street={dummyStreetObject}
        editable
        onClick={handleClick}
      />
    )

    wrapper.simulate('click')
    expect(handleClick).toBeCalled()
  })
})
