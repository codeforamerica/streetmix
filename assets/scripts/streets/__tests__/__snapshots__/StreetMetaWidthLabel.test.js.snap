// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreetMetaWidthLabel renders when editable 1`] = `
<span
  className="street-width street-width-editable"
  onClick={[MockFunction]}
  title="Change width of the street"
>
  <FormattedMessage
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
</span>
`;

exports[`StreetMetaWidthLabel renders when not editable 1`] = `
<span
  className="street-width"
  onClick={[MockFunction]}
  title={null}
>
  <FormattedMessage
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
</span>
`;

exports[`StreetMetaWidthLabel renders with over available width 1`] = `
<span
  className="street-width street-width-editable"
  onClick={[MockFunction]}
  title="Change width of the street"
>
  <FormattedMessage
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
  <span
    className="street-width-over"
  >
    <FormattedMessage
      defaultMessage="({width} over)"
      id="width.over"
      values={
        Object {
          "width": "0.3 m",
        }
      }
    />
  </span>
</span>
`;

exports[`StreetMetaWidthLabel renders with remaining width 1`] = `
<span
  className="street-width street-width-editable"
  onClick={[MockFunction]}
  title="Change width of the street"
>
  <FormattedMessage
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
  <span
    className="street-width-under"
  >
    <FormattedMessage
      defaultMessage="({width} room)"
      id="width.under"
      values={
        Object {
          "width": "0.3 m",
        }
      }
    />
  </span>
</span>
`;
